register-system.yaml

```diff
@@ -189,3 +199,45 @@
       delegate_to: 127.0.0.1
       loop: "{{ ansible_interfaces }}"
       when: interface.type == "ether" and interface.ipv4.address is defined
+
+    - name: "Pause for some time so that lldptool has enough time to capture necessary information"
+      pause:
+        minutes: 5
+
+    - name: "Get switchport name that's connected to NIC"
+      shell: >-
+        lldptool -i {{ item }} get-tlv -n -V portID | awk -F": " '/Ifname|Local/ {print $2}'
+      failed_when: false
+      loop: "{{ physical_interfaces.stdout_lines }}"
+      register: port_ids
+
+    - name: "Get hostname of the switch that a NIC is connected to"
+      shell: >-
+        lldptool -i {{ item.item }} get-tlv -n -V sysName
+      failed_when: false
+      loop: "{{ port_ids.results }}"
+      register: sysnames
+
+    - set_fact:
+        triplets: "{{ triplets + [[item.item.item, item.item.stdout, item.stdout.split('\t')[1] ]] }}"
```

Comment:
I think we should probably use a more meaningful name here. I know this is my fault, but in my defense I was still figuring things out at the time and didn't have brain to spare on naming things :) We should also probably add a `name` to this task.